google-cloud-logging/src/main/java/com/google/cloud/logging/Context.java

```diff
@@ -142,26 +148,15 @@ public Builder loadCloudTraceContext(String cloudTrace) {
      */
     public Builder loadW3CTraceParentContext(String traceParent) throws IllegalArgumentException {
       if (traceParent != null) {
-        String[] fields = traceParent.split("-");
-        if (fields.length > 3) {
-          String versionFormat = fields[0];
-          if (!versionFormat.equals("00")) {
-            throw new IllegalArgumentException("Not supporting versionFormat other than \"00\"");
-          }
-        } else {
+        Matcher validator = W3C_TRACE_CONTEXT_FORMAT.matcher(traceParent.toLowerCase());
```

Comment:
I'm not sure if `toLowerCase()` is needed. The [spec says](https://www.w3.org/TR/trace-context/#parent-id) "Vendors MUST ignore the traceparent when the parent-id is invalid (for example, if it contains non-lowercase hex characters)."

Reply:
I do not know whether we should be so strict about it. A chance that two different trace ids will be produced that differ only by a specific letter register seems low to me. But this way we can avoid exceptions that interrupt the metadata collection in the library.

Reply:
Should we use then toUpperCase() to be safe?

Reply:
My vote would be to just ignore any `traceparent` headers with invalid characters as the spec suggests, that upper-case "MUST" in the docs seems pretty serious haha. 

My interpretation is that header parsing is meant to be farily strict, we're not supposed to alter the fields to make the data fit our expectations. But I'll leave the decision up to you.

Reply:
I agree with you about the standard. My concern is that ignoring it (or throwing and catching exception on the same account) would be very hard to troubleshoot for application developers. To be a little more liberal on this might prove to be closer to the spirit of the shared library.